core/src/test/java/org/elasticsearch/cluster/routing/PrimaryAllocationIT.java

```diff
@@ -103,6 +116,63 @@ public void testDoNotAllowStaleReplicasToBePromotedToPrimary() throws Exception
         assertHitCount(client().prepareSearch().setSize(0).setQuery(matchAllQuery()).get(), 2l);
     }
 
+    public void testFailedAllocationOfStalePrimaryToDataNodeWithNoData() throws Exception {
+        createStaleReplicaScenario();
```

Comment:
I think this can be made simpler by just creating a one node cluster, create an index with no replicas, add a node to the cluster. Shutdown the first node and force assign. Makes sense?